CardiorespiratoryFitness/CardiorespiratoryFitness/iOS/CRFHeartRateRecorder.swift

```diff
@@ -97,21 +97,31 @@ public class CRFHeartRateRecorder : RSDSampleRecorder, CRFHeartRateVideoProcesso
         return highConfidenceSamples.last
     }
     
-    public func vo2Max() -> (value: Double, confidence: Double)? {
-        guard self.bpmSamples.count > 0 else { return nil }
+    public func vo2Max() -> Double? {
+        guard self.bpmSamples.count > 0,
+            let sexValue = self.taskViewModel.taskResult.findAnswerResult(with: CRFDemographicsKeys.sex.stringValue)?.value as? String,
+            let sex = CRFSex(rawValue: sexValue),
+            let birthYear = self.taskViewModel.taskResult.findAnswerResult(with: CRFDemographicsKeys.birthYear.stringValue)?.value as? Int
+            else {
+                return nil
+        }
         let startTime = self.clock.startSystemUptime + 30
         let highConfidenceSamples = self.bpmSamples.filter {
             $0.confidence >= CRFMinConfidence &&
             ($0.timestamp ?? 0) >= startTime
         }
-        guard highConfidenceSamples.count > 1,
-            let first = highConfidenceSamples.first,
-            let last = highConfidenceSamples.last
-            else {
-                return nil
+        guard highConfidenceSamples.count > 1 else { return nil }
+
+        let age = Double(Calendar(identifier: .iso8601).component(.year, from: Date()) - birthYear)
+        let meanHR = highConfidenceSamples.map({ $0.bpm }).mean()
```

Comment:
Did the step test just to see what the results were and compared to my Fitbit scores (from a while back when I was in better shape!) this would be consistent with a half-ing of the actual score resulting from double the beats in the algo
![IMG_1B9E2FB04898-1](https://user-images.githubusercontent.com/3682103/55663653-121cb580-57d6-11e9-99f2-df31567feb57.jpeg)
![IMG_0677](https://user-images.githubusercontent.com/3682103/55663656-1648d300-57d6-11e9-9549-149404cac9f7.PNG)

Reply:
Oops, yes. good catch. thanks!